src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,82 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int ARRAY_SIZE = 10;
+
+    private final Container<K, V>[] containers;
+    private int counter;
+
+    public StorageImpl() {
+        containers = new Container[ARRAY_SIZE];
+        counter = 0;
+
+    }
+
     @Override
     public void put(K key, V value) {
+        if (counter > ARRAY_SIZE) {
+            counter = 0;
+        }
+        if (get(key) != null) {
+            rewriteValueByKey(key, value);
+        } else {
+            containers[counter] = new Container<>(key, value, counter);
+            counter++;
+        }
+    }
+
+    private void rewriteValueByKey(K key, V value) {
+        Container<K, V> lookingContainer = getContainerByKey(key);
```

Comment:
just suggestion, but 
```suggestion
        Container<K, V> existingContainer = getContainerByKey(key);
```